Examples/Detectors/Geant4Detector/src/MockupSectorBuilder.cpp

```diff
@@ -0,0 +1,344 @@
+
+  auto minZ = *min_element(positionZ.begin(), positionZ.end());
+  auto maxZ = *max_element(positionZ.begin(), positionZ.end());
+
+  auto minX = *min_element(positionX.begin(), positionX.end());
+  auto maxX = *max_element(positionX.begin(), positionX.end());
+
+  Acts::ActsScalar hx = straw_surfaces.front()->bounds().values()[1] + 10;
+  Acts::ActsScalar hy = 0.5 * (std::abs(minY - maxY) + 2 * radius + 10.);
+  Acts::ActsScalar hz = 0.5 * (std::abs(minZ - maxZ) + 2 * radius + 10.);
+
+  auto detectorVolumeBounds =
+      std::make_unique<Acts::CuboidVolumeBounds>(hx, hy, hz);
+
+  Acts::Vector3 chamber_position = {(maxX + minX) / 2, (maxY + minY) / 2,
+                                    (maxZ + minZ) / 2};
+
+  // create the detector volume for the chamber
+  auto detectorVolume = Acts::Experimental::DetectorVolumeFactory::construct(
+      Acts::Experimental::defaultPortalGenerator(), gctx, chamber_config.name,
+      Acts::Transform3(Acts::Translation3(chamber_position)),
+      std::move(detectorVolumeBounds), straw_surfaces,
+      std::vector<std::shared_ptr<Acts::Experimental::DetectorVolume>>{},
+      Acts::Experimental::allPortalsAndSurfaces());
+
+  return detectorVolume;
+}
+
+std::shared_ptr<Acts::Experimental::DetectorVolume>
+ActsExamples::MockupSectorBuilder::BuildSector(
+    std::vector<std::shared_ptr<Acts::Experimental::DetectorVolume>>
+        det_volumes,
+    const Acts::GeometryContext& gctx) {
+  if (m_cfg.NumberOfSectors > maxNumberOfSectors) {
+    throw std::invalid_argument("MockupSector:Number of max sectors exceeded");
+    return nullptr;
+  }
+
+  // sort the dtector volumes by their radial distance (from
+  // innermost---->outermost)
+  std::sort(det_volumes.begin(), det_volumes.end(), sortByRadialDistance);
+
+  // calculate the inner and outer radius of the cylinderbounds
+  auto rmin = det_volumes.front()->center().y() -
+              2 * det_volumes.front()->volumeBounds().values()[1];
+  auto rmax = det_volumes.back()->center().y() +
+              3 * det_volumes.back()->volumeBounds().values()[1];
+  auto halfZ = det_volumes.back()->volumeBounds().values()[2];
+
+  // radius of the inner cylinder for the inner chambers
+  auto rmin_inner = rmin;
+  auto rmax_inner = det_volumes.front()->center().y() +
+                    3 * det_volumes.front()->volumeBounds().values()[1];
+
+  // radius of the middle cylinder for the middle chambers
+  auto rmin_middle = det_volumes[1]->center().y() -
+                     2 * det_volumes[1]->volumeBounds().values()[1];
+  auto rmax_middle = det_volumes[1]->center().y() +
+                     3 * det_volumes[1]->volumeBounds().values()[1];
+
+  // radius of the outer cylinder
+  auto rmin_outer = det_volumes.back()->center().y() -
+                    2 * det_volumes.back()->volumeBounds().values()[1];
+  auto rmax_outer = det_volumes.back()->center().y() +
+                    3 * det_volumes.back()->volumeBounds().values()[1];
```

Comment:
The code here, and below, relies on 3 layers (inner, middle and outer) being built. I think we need to generalise this such that adding/ removing one chamber would be possible.

Reply:
Yes i agree, but my question is if we want to insert the cylinder volumes for each of the layers (inner, middle, outer) regardless of whether there are not all the chambers there . because in that case, I need to define somehow the parameters of the cylinder bounds. Also, according to what I have done so far, all the sectors are identical, should I leave it like that? because if all sectors are identical, i think it is better not to have all the cylinder volumes for the layers (inner,middle,outer) (e. g if the inner chamber is absent, the cylinder volume should also absent, too)